### DISCUSSION TITLE
Whats the preferred way to send go routine/background task output to frontend?

### DISCUSSION THREAD

**Discussion by renepardon** [wailsapp/wails#2589] [General]
I want to run multiple background tasks and send the output to the frontend to display those (e.g. cli output/stdout/stderr) and for displaying progress information for long running tasks.

What would be the preferred way? Creating a Websocket server or using IPC or polling through IPC?

**Comment by marcio199226** [comment]
You can check out how I'm doing it in my app 
https://github.com/marcio199226/ytd/blob/v2-wails/app.go#L220
https://github.com/marcio199226/ytd/blob/v2-wails/ngrok.go#L256

There are a couple of other examples in that app where there is an bg task initialized from UI or not and that bg task send events to frontend to notify about any progress.

I don't know if there are better ways to achieve it but for my purpose it's work pretty good

Hope I've helped you

**Reply by renepardon** [reply]
So you just emit events. The websocket server is just for development I guess? It might be too many events in my case. But it’s a good start already, thank you!

**Reply by marcio199226** [reply]
yeah I think is enough once you figure out that it's not enough performant or has other issues you could always implement some other methods or dispatch events in batch
